src/views/admin/PersonDetailsView.vue

```diff
@@ -152,38 +153,52 @@
     await personStore.deletePersonById(personId);
   }
 
-  function keyMapper(key: string): string {
-    switch (key) {
-      case LockKeys.LockedFrom:
-        return t('person.lockedBy');
-      case LockKeys.Timestamp:
-        return t('since');
-      default:
-        return key;
-    }
-  }
-
-  function keyValueMapper(key: string, value: string): string {
-    if (key === LockKeys.Timestamp) {
-      return new Intl.DateTimeFormat('de-DE', {
-        year: 'numeric',
-        month: '2-digit',
-        day: '2-digit',
-      }).format(new Date(value));
-    }
-    return value;
+  function getOrganisationDisplayName(organisation: Organisation): string {
+    return organisation.kennung ? `${organisation.kennung} (${organisation.name})` : organisation.name;
   }
 
+  // translate keys and format attributes for display
   const getLockInfo: ComputedRef<{ key: string; attribute: string }[]> = computed(() => {
     if (!personStore.currentPerson?.person.isLocked) return [];
+
     const { lockInfo }: Person = personStore.currentPerson.person;
     if (!lockInfo) return [];
-    return Object.entries(lockInfo).map(([key, value]: [string, string]) => ({
-      key: keyMapper(key),
-      attribute: keyValueMapper(key, value.toString()),
-    }));
+
+    return Object.entries(lockInfo).map(([key, attribute]: [string, string]) => {
+      switch (key) {
+        case LockKeys.LockedFrom:
+          return {
+            key: t('person.lockedBy'),
+            attribute: organisationStore.lockingOrganisation
+              ? getOrganisationDisplayName(organisationStore.lockingOrganisation)
+              : t('admin.organisation.unknownOrganisation'),
+          };
+
+        case LockKeys.Timestamp:
+          return {
+            key: t('since'),
+            attribute: new Intl.DateTimeFormat('de-DE', {
+              year: 'numeric',
+              month: '2-digit',
+              day: '2-digit',
+            }).format(new Date(attribute)),
+          };
+
+        default:
+          return { key, attribute };
+      }
+    });
   });
 
+  watch(
+    () => personStore.currentPerson,
+    async () => {
+      if (!personStore.currentPerson?.person.isLocked) return;
+      if (!personStore.currentPerson.person.lockInfo) return;
+      await organisationStore.getLockingOrganisationById(personStore.currentPerson.person.lockInfo.lock_locked_from);
+    },
+  );
```

Comment:
```suggestion
watch(
  () => personStore.currentPerson?.person,
  async (person) => {
    const lockInfo = person?.lockInfo;
    if (person?.isLocked && lockInfo) {
      await organisationStore.getLockingOrganisationById(lockInfo.lock_locked_from);
    }
  }
);
```

Weniger code, gleiches Ergebnis vielleicht?

Reply:
Ich habe das Destructuring übernommen und die Bedingung zu einer zusammengefasst. Passt das so?